src/fees/compute_fees.py

```diff
@@ -62,20 +84,38 @@ def surplus(self) -> int:
             return current_limit_sell_amount - self.sell_amount
         raise ValueError(f"Order kind {self.kind} is invalid.")
 
-    def raw_surplus(self) -> int:
-        """Compute raw surplus of a trade in the surplus token
-        First, the application of protocol fees is reversed. Then, surplus of the resulting trade
-        is computed."""
+    def compute_all_fees(self) -> None:
         raw_trade = deepcopy(self)
-        for fee_policy in reversed(self.fee_policies):
-            raw_trade = fee_policy.reverse_protocol_fee(raw_trade)
-        return raw_trade.surplus()
+        i = 0
+        self.protocol_fee = 0
+        self.partner_fee = 0
+        if self.fee_policies:
```

Comment:
Since the old comment was removed: This line can potentially be deleted.

Reply:
There is line 98 that needs to execute after the for-loop finishes, but it only makes sense if there was at least one fee policy.

Reply:
Oh, i see. But to me it is simpler to write
```python
        partner_fee = 0
        for i, fee_policy in enumerate(reversed(self.fee_policies)):
            raw_trade = fee_policy.reverse_protocol_fee(raw_trade)
            ## we assume that partner fee is the last to be applied
            if i == 0 and self.partner_fee_recipient is not NULL_ADDRESS:
                partner_fee = raw_trade.surplus() - self.surplus()
        total_protocol_fee = raw_trade.surplus() - self.surplus()
```
The other two initializations are not required.

That part of the code might even benefit from a function
```python
    def partner_fee(self):
        if not self.fee_policies or self.partner_fee_recipient is NULL_ADDRESS:
            return 0
        # we assume that partner fee is the last to be applied
        fee_policy = self.fee_policies[-1]
        raw_trade = deepcopy(self)
        raw_trade = fee_policy.reverse_protocol_fee(raw_trade)
        return raw_trade.surplus() - self.surplus()
```
and then just
```python
        partner_fee = self.partner_fee()
        total_protocol_fee = self.total_protocol_fee()
```
The small loss in efficiency should not matter compared to the increase in readability. (Though you might disagree on the readability, then feel free to ignore).

Reply:
Actually you are right and my concern was not really justified. I simplified the code and followed your first suggestion.